### PR TITLE
feat(user login): implement token response on successful login

### PR DESCRIPTION
**Pull request by FaithAdekunle**
#### What does this PR do?
Sends user a JWT upon successful login
Displays an error feedback upon failed signup/login attempt

#### Description of Task to be completed?
- Send a JWT on successful login
- Handle login and signup errors
- Ensure users can't signup with existing email or username

#### How should this be manually tested?
- Run `git clone https://github.com/andela/nazgul-cfh.git` to clone the repository.
- Navigate to the nazgul-cfh folder.
- Install dependencies with `npm install`
- Install and run mongodb
- Create a `.env` file and add fields to it following the directive in the .env.example file
- run `npm start:dev`
- Navigate to `localhost:3000` in your browser

#### What are the relevant pivotal tracker stories?
#156100321
#### Screenshots (if appropriate)
image
image
image

### CODE REVIEW

**Comment by FaithAdekunle** [comment]
Hey guys @iidrees @topseySuave @Felglitzzz @Efosaok @omokolataiwo 
Please could you take a look at my PR and give give me feedback.
Thanks.

**Review by iidrees — COMMENTED** [review]
Great PR @FaithAdekunle ,
But I noticed that the signin  and signup HTML files have strange indentations.
It would be more readable if you made the indentation uniform.
Aside that, great job 👍 .

File: test/user/signup.js [hunk 1]
```diff
@@ -105,15 +105,95 @@ describe('Signup API', () => {
...
       .send({
         name: 'Full name',
-        email: 'testtt@test.com',
-        username: 'user',
+        email: 'tester@test.com',
+        username: 'user_test',
         password: 'password'
       })
       .end((err, res) => {
+        // console.log(res);
```
Comment (morekoya) [inline test/user/signup.js]: I would suggest leaving out  redundant code that has been commented out

File: test/user/signup.js [hunk 2]
```diff
@@ -105,15 +105,95 @@ describe('Signup API', () => {
...
         expect(res.status).to.equal(201);
-        expect(err).to.be.null;
         expect(res.body).to.be.an('object');
         done();
       });
   });
 });
+
+describe('Signin', () => {
+  it('should not allow a user to login with no email', (done) => {
```
Comment (morekoya) [inline test/user/signup.js]: I thought these test were for signup? I would expect the test description to match.

**Comment by topseySuave** [comment]
Great work @FaithAdekunle, I also noticed the signup tests have a description about signing in, I believe that test portion can be modified adequately, to fit the appropriate test scenario. Great work on the sign in feature.

**Comment by Efosaok** [comment]
Good job for making sure to write tests for this feature,
though this is not a big deal, It is more accepted or a general convention that when writing tests for api endpoints,the describe blocks description should follow this format `<request method in uppercase>  <api url>` something like `POST api/users/signin` ,this simple but generic message conveys more information about that particular test-suite than just saying *Signin* .

Great job once again

File: test/user/signup.js (hunk 1 above)
Comment (FaithAdekunle) [reply test/user/signup.js]: Thanks for pointing this out. The unused line has been removed.

File: test/user/signup.js (hunk 2 above)
Comment (FaithAdekunle) [reply test/user/signup.js]: Thanks for your feedback. The test descriptions and file name have been refactored to match.

**Comment by FaithAdekunle** [comment]
@topseySuave @Efosaok @iidrees thanks for your feedback guys. They have been acknowledged and worked on.

**Pull request merged**